src/CoreWCF.WebHttp/src/CoreWCF/UriTemplateLiteralPathSegment.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Specialized;
+using System.Text;
+using CoreWCF.Runtime;
+
+namespace CoreWCF
+{
+    // thin wrapper around string; use type system to help ensure we
+    // are doing canonicalization right/consistently
+    internal class UriTemplateLiteralPathSegment : UriTemplatePathSegment, IComparable<UriTemplateLiteralPathSegment>
+    {
+
+        // segment doesn't store trailing slash
+        private readonly string _segment;
+        private static readonly Uri s_dummyUri = new Uri("http://localhost");
+
+        UriTemplateLiteralPathSegment(string segment)
+            : base(segment, UriTemplatePartType.Literal, segment.EndsWith("/", StringComparison.Ordinal))
```

Comment:
Missing accessibility modifier

Reply:
This is addressed in cdbaeef71cac50fe69f7dc1d2722948f430e6465.